api/server/sdk/credentials_test.go

```diff
@@ -1099,3 +1099,80 @@ func TestSdkCredentialGetOwnershipFromCred(t *testing.T) {
 	assert.NoError(t, err)
 	assert.NotNil(t, o)
 }
+
+func TestSdkCredentialRemovereferencesSuccess(t *testing.T) {
+
+	// Create server and client connection
+	s := newTestServer(t)
+	defer s.Stop()
+
+	uuid := "good-uuid"
+
+	req := &api.SdkCredentialRemoveReferencesRequest{CredentialId: uuid}
+
+	enumAzure := map[string]interface{}{
+		api.OptCredType:             "azure",
+		api.OptCredAzureAccountName: "test-azure-account",
+		api.OptCredAzureAccountKey:  "test-azure-account",
+		api.OptCredProxy:            "false",
+		api.OptCredIAMPolicy:        "false",
+	}
+	enumerateData := map[string]interface{}{
+		uuid: enumAzure,
+	}
+	s.MockDriver().
+		EXPECT().
+		CredsEnumerate().
+		Return(enumerateData, nil)
+	s.MockDriver().
+		EXPECT().
+		CredsRemoveReferences(uuid).
+		Return(nil)
+
+	// Setup client
+	c := api.NewOpenStorageCredentialsClient(s.Conn())
+
+	// Validate Created Credentials
+	_, err := c.RemoveReferences(context.Background(), req)
+	assert.NoError(t, err)
+}
+
+func TestSdkCredentialRemoveReferencesFailed(t *testing.T) {
+
+	// Create server and client connection
+	s := newTestServer(t)
+	defer s.Stop()
+
+	uuid := "bad-uuid"
+
+	req := &api.SdkCredentialRemoveReferencesRequest{CredentialId: uuid}
+	enumAzure := map[string]interface{}{
+		api.OptCredType:             "azure",
+		api.OptCredAzureAccountName: "test-azure-account",
+		api.OptCredAzureAccountKey:  "test-azure-account",
+		api.OptCredProxy:            "false",
+		api.OptCredIAMPolicy:        "false",
+	}
+	enumerateData := map[string]interface{}{
+		uuid: enumAzure,
+	}
+
+	s.MockDriver().
+		EXPECT().
+		CredsEnumerate().
+		Return(enumerateData, nil)
+	s.MockDriver().
+		EXPECT().
+		CredsRemoveReferences(uuid).
+		Return(fmt.Errorf("Failed to remove refs"))
```

Comment:
I did not clearly get the difference between the TestSdkCredentialRemovereferencesSuccess() and TestSdkCredentialRemoveReferencesFailed()

Only the uuid is different and the mock driver return code is different. If the mock driver is not going to consult the enumerateData, then why even populate it?

Reply:
The goal is to test error response for the bad case.  We dont need to populate the data as such. but does not hurt.